### PR TITLE
Feature : Sync Auto Reaction To Scrapbook Website

### PR DESCRIPTION
**Pull request by DevIos01**
In Issue [#670](https://github.com/hackclub/scrapbook/issues/670) on the Hack Club Scrapbook GitHub, as suggested to make sure that when Scrappy automatically adds reactions, these same reactions show up on the website too. This way, everything matches and is automatically labeled without users having to add emojis again by themselves.

Now, I have made it so the emojis that Scrappy picks out automatically get added to our messages database. They're also shown and updated on the Scrapbook website, making things more consistent and easier for everyone.

### CODE REVIEW

File: src/lib/reactions.js [hunk 1]
```diff
@@ -10,18 +10,31 @@ export const getReactionRecord = async (emoji, updateId) =>
...
 
-export const reactBasedOnKeywords = (channel, message, ts) => {
-  Object.keys(emojiKeywords).forEach(async (keyword) => {
-    try {
-       if (
-        message?.toLowerCase().search(new RegExp("\\b" + keyword + "\\b", "gi")) !== -1
-      ) {
+export const reactBasedOnKeywords = async (channel, message, ts) => {
+  for (const [keyword, emojiName] of Object.entries(emojiKeywords)) {
+    if (message?.toLowerCase().includes(keyword.toLowerCase())) {
```
Comment (JosiasAurel) [inline src/lib/reactions.js:15]: you sure this regex is equivalent?

File: src/lib/streaks.js [hunk 2]
```diff
@@ -46,6 +47,28 @@ export const incrementStreakCount = (userId, channel, message, ts) =>
       "https://scrapbook.hackclub.com/" + userRecord.username;
     await react("remove", channel, ts, "beachball"); // remove beachball react
     await react("add", channel, ts, SEASON_EMOJI);
+    console.log(`Added ${SEASON_EMOJI} reaction to message ${ts} in channel ${channel}`);
```
Comment (JosiasAurel) [inline src/lib/streaks.js:50]: let's remove this log

File: src/events/reactionAdded.js [hunk 3]
```diff
@@ -33,6 +33,28 @@ export default async ({ event }) => {
...
+    const emojiRecord = await getEmojiRecord(SEASON_EMOJI);
+    const update = await prisma.updates.findFirst({
+      where: {
+        messageTimestamp: parseFloat(ts),
+      },
+    });
+
+    if (update) {
+      const reactionExists = await emojiExists(SEASON_EMOJI, update.id);
+      console.log(`Does emoji reaction exist for ${SEASON_EMOJI}?`, reactionExists);
```
Comment (JosiasAurel) [inline src/events/reactionAdded.js:47]: remove this log

**Review by JosiasAurel — CHANGES_REQUESTED** [review]
please remove the logs... also is there a way to not repetitively get those updates/reactions?

File: src/lib/slack.js [hunk 4]
```diff
@@ -9,9 +11,38 @@ export const react = async (addOrRemove, channel, ts, reaction) => {
...
+        await prisma.emojiReactions.deleteMany({
+          where: {
+            updateId: update.id,
+            emojiTypeName: emojiRecord.name,
+          },
+        });
+      }
+    }
+
+    metrics.increment(`success.react`, 1);
```
Comment (JosiasAurel) [inline src/lib/slack.js:43]: please specify whether or not we're adding or removing the reaction using those template literals

**Review by JosiasAurel — CHANGES_REQUESTED** [review]

**Review by JosiasAurel — APPROVED** [review]

**Review by grymmy — APPROVED** [review]

**Pull request merged**